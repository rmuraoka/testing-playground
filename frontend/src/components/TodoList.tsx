import React, {useEffect, useState} from 'react';
import {Box, Button, Typography} from '@mui/material';
import TodoForm from "./TodoForm";
import TodoItem from "./TodoItem";

// Todo型の定義
interface Todo {
    id: number;
    title: string;
    description: string;
}

interface TodoListProps {
    todos: Todo[];
    onUpdateClick: (id: number, title: string, description: string) => void; // 更新処理のためのプロパティ
}

// TodoListコンポーネント
const TodoList: React.FC<TodoListProps> = ({ todos, onUpdateClick }) => { // 修正: 引数をオブジェクトで受け取る
    return (
        <>
            <Box>
                {todos.map((todo) => (
                    <TodoItem
                        id={todo.id}
                        title={todo.title}
                        description={todo.description}
                        onUpdate={onUpdateClick}
                    />
                ))}
            </Box>
        </>
    );
};

// TodoListContainerコンポーネント
export const TodoListContainer: React.FC = () => {
    const [todos, setTodos] = useState([]);
    const fetchTodos = async () => {
        fetch(`${process.env.REACT_APP_API_URL}/todos`)
            .then((response) => response.json())
            .then((data) => setTodos(data))
            .catch((error) => console.error('Error fetching todos:', error));
    };

    const handleAddTodo = async (newTodo: { title: string; description?: string }) => {
        const response = await fetch(`${process.env.REACT_APP_API_URL}/todos`, {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify(newTodo),
        });

        if (response.ok) {
            fetchTodos();
        } else {
            console.error('Failed to add todo:', response.statusText);
        }
    };

    const handleUpdateTodo = async (id: number, title: string, description: string) => {
        const response = await fetch(`${process.env.REACT_APP_API_URL}/todos/${id}`, {
            method: 'PUT',
            headers: {
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({ title, description }),
        });
        if (response.ok) {
            fetchTodos(); // 更新後にタスクを再取得
        }
    };

    useEffect(() => {
        fetchTodos(); // 初回マウント時にタスクを取得
    }, []);

    return (
        <Box>
            <Typography variant="h4" gutterBottom>
                Todo List
            </Typography>
            <TodoForm onAddTodo={handleAddTodo} />
            <TodoList todos={todos} onUpdateClick={handleUpdateTodo}/>
        </Box>
    );
};

export default TodoList;

