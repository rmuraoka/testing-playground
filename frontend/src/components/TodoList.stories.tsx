import React from 'react';
import {Meta} from '@storybook/react';
import TodoList from "./TodoList";

const sampleTodos = [
    { id: 1, title: 'Sample Task 1', description: 'Description for task 1' },
    { id: 2, title: 'Sample Task 2', description: 'Description for task 2' },
];

export default {
    title: 'TodoList',
    component: TodoList,
} as Meta;

export const Default = () => <TodoList todos={sampleTodos}/>;
