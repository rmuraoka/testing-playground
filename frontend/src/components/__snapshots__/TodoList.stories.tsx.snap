// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Default smoke-test 1`] = `
<div class="MuiBox-root css-0">
  <div class="MuiBox-root css-mkpezx">
    <div>
      <h6 class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root">
        Sample Task 1
      </h6>
      <p class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root">
        Description for task 1
      </p>
      <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary css-1x6kmiy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
              data-testid="delete-task-button-Sample Task 1"
      >
        Delete
      </button>
    </div>
  </div>
  <div class="MuiBox-root css-mkpezx">
    <div>
      <h6 class="MuiTypography-root MuiTypography-h6 css-1miy0lu-MuiTypography-root">
        Sample Task 2
      </h6>
      <p class="MuiTypography-root MuiTypography-body2 css-1a1whku-MuiTypography-root">
        Description for task 2
      </p>
      <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorSecondary css-1x6kmiy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
              data-testid="delete-task-button-Sample Task 2"
      >
        Delete
      </button>
    </div>
  </div>
</div>
`;
